src/core/src/package_managers/ZypperPackageManager.py

```diff
@@ -44,71 +44,86 @@ def __init__(self, env_layer, execution_config, composite_logger, telemetry_writ
         # Package manager exit code(s)
         self.zypper_exitcode_ok = 0
         self.zypper_exitcode_zypper_updated = 103
+        self.zypper_exitcode_zypp_locked = 7
 
         # Support to check for processes requiring restart
         self.zypper_ps = "sudo zypper ps -s"
 
         # Miscellaneous
         self.set_package_manager_setting(Constants.PKG_MGR_SETTING_IDENTITY, Constants.ZYPPER)
         self.zypper_get_process_tree_cmd = 'ps --forest -o pid,cmd -g $(ps -o sid= -p {})'
+        self.package_manager_max_retries = 5
+        self.error_codes_to_retry = [self.zypper_exitcode_zypp_locked]
 
     def refresh_repo(self):
         self.composite_logger.log("Refreshing local repo...")
         # self.invoke_package_manager(self.repo_clean)  # purges local metadata for rebuild - addresses a possible customer environment error
-        for i in range(0, Constants.MAX_ZYPPER_REPO_REFRESH_RETRY_COUNT):
-            try:
-                self.invoke_package_manager(self.repo_refresh)
-                return
-            except Exception as error:
-                if i < Constants.MAX_ZYPPER_REPO_REFRESH_RETRY_COUNT - 1:
-                    self.composite_logger.log_warning("Exception on package manager refresh repo. [Exception={0}] [RetryCount={1}]".format(repr(error), str(i)))
-                    time.sleep(pow(2, i) + 1)
-                else:
-                    if Constants.ERROR_ADDED_TO_STATUS in repr(error):
-                        error.args = error.args[:1]  # remove Constants.ERROR_ADDED_TO_STATUS flag to add new message to status
-
-                    error_msg = "Unable to refresh repo (retries exhausted). [{0}] [RetryCount={1}]".format(repr(error), str(i))
-
-                    # Reboot if not already done
-                    if self.status_handler.get_installation_reboot_status() == Constants.RebootStatus.COMPLETED:
-                        error_msg = "Unable to refresh repo (retries exhausted after reboot). [{0}] [RetryCount={1}]".format(repr(error), str(i))
-                    else:
-                        self.composite_logger.log_warning("Setting force_reboot flag to True.")
-                        self.force_reboot = True
-                        
-                    self.composite_logger.log_warning(error_msg)
-                    self.status_handler.add_error_to_status(error_msg, Constants.PatchOperationErrorCodes.PACKAGE_MANAGER_FAILURE)
-
-                    raise Exception(error_msg)
+        try:
+            self.invoke_package_manager(self.repo_refresh)
+            return
```

Comment:
minor: don't need return here, function should return if there is no exception and invoke_package_manager completes

Reply:
Right, fixed